geo-types/src/coordinate.rs

```diff
@@ -25,11 +26,25 @@ use approx::{AbsDiffEq, RelativeEq, UlpsEq};
 /// [vector space]: //en.wikipedia.org/wiki/Vector_space
 #[derive(Eq, PartialEq, Clone, Copy, Debug, Hash, Default)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub struct Coordinate<T: CoordNum> {
+pub struct CoordTZM<T: CoordNum, Z: ZCoord, M: Measure> {
     pub x: T,
     pub y: T,
+    pub z: Z,
+    pub m: M,
 }
 
+impl<T: CoordNum, Z: ZCoord, M: Measure> CoordTZM<T, Z, M> {
+    pub fn new(x: T, y: T, z: Z, m: M) -> Self {
+        Self { x, y, z, m }
+    }
+}
+
+pub type Coordinate<T> = CoordTZM<T, NoValue, NoValue>;
```

Comment:
One downside of the changes in this pull request is if the user wants to view the documentation for `Coordinate` (or `Coord`, `CoordM`, `CoordZ`, or `CoordZM`), they'll need to click through to `CoordTZM` to actually see it
<img width="1287" alt="Screen Shot 2022-04-09 at 12 55 06 AM" src="https://user-images.githubusercontent.com/416575/162557075-b48bdfb1-5be4-470c-a0b1-1861be0ce554.png">

<img width="1275" alt="Screen Shot 2022-04-09 at 12 55 15 AM" src="https://user-images.githubusercontent.com/416575/162557078-d74d3aea-8514-4d48-ad19-bb053a318bf7.png">

One option here is to move the documentation to a separate Markdown file, and then we can add `#[doc = include_str!("../docs/coord.md")]` above each type alias here. Though the downside there is all the docs will be the same, which we may not want.

@nyurik What do you think?

Reply:
I think I will add proper documentation for each variant, with appropriate examples for each. Variant's description might be shorter than the main one, with a link to the main.

Reply:
@frewsxcv thx for the feedback - I added some documentation to the variants.  I think we may want to do another PR with more doc fixes, or it might be too big and will make reviewing even harder.